### PR TITLE
sql: pg_builtins.proargtypes is an oid[]

### PR DESCRIPTION
**Pull request by jordanlewis**
Previously, the `proargtypes` field of the `pg_builtins` table was a
string type. It's supposed to be an `oidvector`.

Release note: None

### CODE REVIEW

File: pkg/sql/pg_catalog.go [hunk 1]
```diff
@@ -1166,7 +1165,7 @@ CREATE TABLE pg_catalog.pg_proc (
 	pronargs INT,
 	pronargdefaults INT,
 	prorettype OID,
-	proargtypes STRING,
+	proargtypes OID[],
 	proallargtypes STRING[],
```
Comment (knz) [inline pkg/sql/pg_catalog.go]: I just checked https://www.postgresql.org/docs/9.6/static/catalog-pg-proc.html

`proallargtypes` should be an `oid[]` ,and `proargtypes` should be an `oidvector`.

**Comment by jordanlewis** [comment]
Thanks for catching that. I went ahead and added the `oidvector` type alias, which is actually important because it has a different output format than `oid[]`.

**Comment by nvanbenschoten** [comment]
:lgtm: nice! Are there any other `STRINGs` in our implementation of the `pg_catalog` which should be `OIDVECTORs`?

---

Reviewed 16 of 16 files at r2, 3 of 3 files at r3.
Review status: all files reviewed at latest revision, 1 unresolved discussion, some commit checks pending.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/23710#-:-L7Q-YQxkbdWuxGqWhFc:bzdqaxt)*

**Comment by knz** [comment]
:lgtm: with nit

---

Reviewed 16 of 16 files at r2, 3 of 3 files at r3.
Review status: all files reviewed at latest revision, 2 unresolved discussions, all commit checks successful.

---

*[pkg/sql/pg_catalog.go, line 1224 at r3](https://reviewable.io:443/reviews/cockroachdb/cockroach/23710#-L7Q2dZ1uAvKhm9739yX:-L7Q2dZ27zVUGwRXtbQv:b-5aslla) ([raw file](https://github.com/cockroachdb/cockroach/blob/fad233c40e01b19331a14f31a3d6e7a6e789a8b7/pkg/sql/pg_catalog.go#L1224)):*
> ```Go
> 
> 					argTypes := builtin.Types
> 					dArgTypes := tree.NewDArray(types.Oid)
> ```

nit: If there was a way to pass a capacity or allocate with a fixed length, that's the place to use it.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/23710#-:-L7Q2lU75V8CevhZsHgQ:b-xv64lu)*

**Comment by jordanlewis** [comment]
TFTRs!

> Are there any other STRINGs in our implementation of the pg_catalog which should be OIDVECTORs?

There are a couple - `indcollation` and `indclass` should be `OIDVECTOR` and `indoption` should be `INT2VECTOR`. Currently they're completely botched (returning a single `0` value) which will trip up every driver that actually looks at those columns. I'll send a separate PR for that later, but it's not high priority.

---

Review status: all files reviewed at latest revision, 2 unresolved discussions, all commit checks successful.

---

*[pkg/sql/pg_catalog.go, line 1169 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/23710#-L7MBQEhGVDRID1Z3ael-r1-1169:-L7Q6hnpEmfqxvcHt7yn:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/fbd6816edc4498254c37f3591911dac5def34741/pkg/sql/pg_catalog.go#L1169)):*
<details><summary><i>Previously, knz (kena) wrote…</i></summary><blockquote>

I just checked https://www.postgresql.org/docs/9.6/static/catalog-pg-proc.html

`proallargtypes` should be an `oid[]` ,and `proargtypes` should be an `oidvector`.
</blockquote></details>

Done.

---

*[pkg/sql/pg_catalog.go, line 1224 at r3](https://reviewable.io:443/reviews/cockroachdb/cockroach/23710#-L7Q2dZ1uAvKhm9739yX:-L7Q6jYoPHPgHOMZA1ps:b-dhmdm0) ([raw file](https://github.com/cockroachdb/cockroach/blob/fad233c40e01b19331a14f31a3d6e7a6e789a8b7/pkg/sql/pg_catalog.go#L1224)):*
<details><summary><i>Previously, knz (kena) wrote…</i></summary><blockquote>

nit: If there was a way to pass a capacity or allocate with a fixed length, that's the place to use it.
</blockquote></details>

Let's fix these all at once whenever we decide to optimize the catalog implementations.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/23710#-:-L7Q6nMtoo5NdzWHiTjl:bo3t9xx)*

**Comment by jordanlewis** [comment]
```
jordan=# select attrelid::regclass || '.' || attname, atttypid::regtype from pg_attribute where atttypid::regtype::text in ('int2vector', 'oidvector');
                  ?column?                  |  atttypid
--------------------------------------------+------------
 pg_proc.proargtypes                        | oidvector
 pg_index.indkey                            | int2vector
 pg_index.indcollation                      | oidvector
 pg_index.indclass                          | oidvector
 pg_index.indoption                         | int2vector
 pg_statistic_ext.stxkeys                   | int2vector
 pg_trigger.tgattr                          | int2vector
 pg_partitioned_table.partattrs             | int2vector
 pg_partitioned_table.partclass             | oidvector
 pg_partitioned_table.partcollation         | oidvector
 pg_proc_proname_args_nsp_index.proargtypes | oidvector
```

**Pull request merged**